pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader.go

```diff
@@ -130,18 +130,25 @@ func DecodeVersionedBytes(res any, vData *pb.VersionedBytes) error {
 	return nil
 }
 
-func (c *Client) GetLatestValue(ctx context.Context, contractName, method string, params, retVal any) error {
+func (c *Client) GetLatestValue(ctx context.Context, contractName, method string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) error {
```

Comment:
Out of scope I guess, but why do we have a separate `package primitives`?

Reply:
Because things like `ConfidenceLevel`, `Block`, `TxHash`... don't fit under query package but they are primitives used for querying so they live under `pkg/types/query/primitives/primitives.go`. Also some of the structs aren't just used for querying, confidenceLevels are used in a bunch of other places, so having to use it with query.ConfidenceLevel would be weird.
Do you have an alternative?

Reply:
It sounds like some things could be in `package query`, but other things would not make sense there. If so, why are we grouping them together in the first place? What about moving the query-specific things to `package query`, and then doing something else with the rest?

Reply:
BCF-3295